### DISCUSSION TITLE
DBSTREAM -  questions on learn_one(x) / predict_one(x)?

### DISCUSSION THREAD

**Discussion by kmsnyde** [online-ml/river#823] [Q&A] [answered]
Hi - I'm new to River and basically to clustering streaming data, as well.  What I'm looking at using is DBSTREAM.  In most examples in the River documentation, a small set of data (X) is put in to a "for loop" and learn_one(x) is performed on each.  Afterward, predict_one([a, b]) is performed on new data.

If I start performing DBSTREAM on a new data stream, is it correct to perform both learn_one(x), followed by predict_one(x) for each item in X, within the "for loop"?  Or do I have to perform a certain amount of learn_one(x) methods to sort of "prime" the algorithm before using predict_one()?  I would appreciate anyone's extended explanation.  Thanks.

**Comment by MaxHalford** [comment, marked as answer]
> If I start performing DBSTREAM on a new data stream, is it correct to perform both learn_one(x), followed by predict_one(x) for each item in X, within the "for loop"?

That's actually the classical way of going online learning: predict followed by fit. It simulates a real-world situation where predictions need to be made before learning is possible.

>  Or do I have to perform a certain amount of learn_one(x) methods to sort of "prime" the algorithm before using predict_one()?

That's essentially "warming up". It's quite a common practice :)

The nice thing is that you can do both. It really depends on what the specifics of your problem.

**Reply by kmsnyde** [reply]
Thanks for you quick feedback on my question.
